schemas/deviceToCloud/gnss/gnss.json

```diff
@@ -0,0 +1,100 @@
+{
+   "title": "GNSS",
+   "description": "GNSS device coordinate data",
+   "type": "object",
+   "oneOf": [
+      {
+         "properties": {
+            "appId": {
+               "type": "string",
+               "const": "GNSS"
+            },
+            "messageType": {
+               "type": "string",
+               "const": "DATA"
+            },
+            "data": {
+               "type": "string",
+               "description": "NEMA string",
+               "pattern": "^\\$(GPGGA|GPGLL|GPGSA|GPGSV|GPRMC|GPVTG)(,[0-9A-Z\\*\\.]*)+\n?$"
+            },
+            "ts": {
+               "type": "number",
+               "description": "Unix timestamp given in milliseconds when the data was sampled"
+            },
+            "time": {
+               "type": "number",
+               "description": "The same as the 'ts' property. Used for backwards compatibility for some firmware versions of ATv2. Please use the 'ts' field instead"
+            }
+         },
+         "additionalProperties": false
+      },
+      {
+         "properties": {
```

Comment:
**problem:** all properties are optional.

Reply:
The `required` prop at the bottom covers both items in the `oneOf` options array.